### PR TITLE
[wip] enable work in server side

### PR DESCRIPTION
**Pull request by Guria**
POC, not to merge

### CODE REVIEW

**Comment by Guria** [comment]
@christianalfoni what do you say?

File: index.js [hunk 1]
```diff
@@ -127,7 +133,7 @@ function router (controller, routes, options) {
 
   }
 
-  router.trigger = function () {
+  router.trigger = function (url) {
```
Comment (Guria) [inline index.js]: maybe we should use separate method since this once check for stored signals

File: index.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 var urlMapper = require('url-mapper');
-var addressbar = require('addressbar');
 var pathToRegexp = require('path-to-regexp');
+try {
+  var addressbar = require('addressbar');
+} catch(e) {}
```
Comment (christianalfoni) [inline index.js]: Not sure what this is for? Is it to handle failing registration of events etc. inside addressbar? Running tests?

Comment (Guria) [inline index.js]: `addressbar` import fails because of running in node env where window is absent.

**Comment by Guria** [comment]
rebased to latest master

File: index.js (hunk 1 above)
Comment (Guria) [inline index.js]: Seems it is not an issue since controller checks if already remembered initial.

File: index.js (hunk 2 above)
Comment (christianalfoni) [inline index.js]: Ah, you thinking about preparing for server... you know, I am actually not sure that the router has any purpose on the server. Lets say you want to open a list of todos, you go to: `/todos`. What you want to happen is this:

```js
app.get('/todos', (req, res) => {
  db.getTodos()
    .then((todos) => {
      const state = Object.assign({}, initialState, {todos: todos});
      const controller = CerebralDummyController(state);
      const appHtml = renderToString(<Container controller={controller}><App/></Container>);
      res.render('index', {app: appHtml, state: state});
    })
});
```

So you do not care about the router in the client, you just care about the request to the server. That request should prepare some data that should be merged into the initial state of the application. Then you create a dummy controller to expose that state to the app when rendered. Then you have to inject the app and actually also the state into the html file. That way the client will render the app correctly, but also build up the initial client side state store correctly with the state from the server.

I think firing off signals etc. on the server will just be confusing. Just think about the components and inject the state they need.

**Comment by Guria** [comment]
@christianalfoni https://github.com/christianalfoni/cerebral-router/pull/49#discussion_r44003889
> Ah, you thinking about preparing for server... you know, I am actually not sure that the router has any purpose on the server. Lets say you want to open a list of todos, you go to: /todos. What you want to happen is this:
```js
app.get('/todos', (req, res) => {
  db.getTodos()
    .then((todos) => {
      const state = Object.assign({}, initialState, {todos: todos});
      const controller = CerebralDummyController(state);
      const appHtml = renderToString(<Container controller={controller}><App/></Container>);
      res.render('index', {app: appHtml, state: state});
    })
});
```
So you do not care about the client router, you just care about the request to the server. That request should prepare some data that should be merged into the initial state of the application. Then you create a dummy controller to expose that state to the app when rendered. Then you have to inject the app and actually also the state into the html file. That way the client will render the app correctly, but also build up the initial client side state store correctly with the state from the server.

>I think firing off signals etc. on the server will just be confusing. Just think about the components and inject the state they need.

I am sure it could be still useful. It will allow not to duplicate routing logic. Preparing inittialState could be achieved with universal actions and we just export state store. Not sure how it would look like, thought. So I'll just leave it here. Maybe sometimes someone will figure out hot it should be.

**Comment by Guria** [comment]
@christianalfoni it proved to be usefull for running signal test in node environment. So I have updated it and merged

**Pull request merged**